### PR TITLE
Call out pg discrepancy with byte->text conversion

### PR DESCRIPTION
**Pull request by justinj**
Added to the "porting from postgres" page.

### CODE REVIEW

**Comment by mjibson** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1925#-:-Ku_pqzeStc2LT5cbsfy:bnfp4nl)*

File: v1.0/porting-postgres.md [hunk 1]
```diff
@@ -96,3 +96,36 @@ In CockroachDB, no such modulo is performed.
...
+----------------------------
+ \x48656c6c6f20776f726c6421
+~~~
+
+In CockroachDB, this conversion is performed by attempting to convert the
+byte data directly into a string:
+
+~~~ sql
+SELECT '\x48656c6c6f20776f726c6421'::BYTEA::TEXT;
++---------------------------------------------+
```
Comment (jseldess) [inline v1.0/porting-postgres.md]: nit: In cockroach examples, we tend to break the command and the response into separate code blocks.

File: v1.0/porting-postgres.md [hunk 2]
```diff
@@ -96,3 +96,36 @@ In CockroachDB, no such modulo is performed.
...
+(1 row)
+~~~
+
+A consequence of this is that such conversions will fail if the byte sequence
+contains invalid UTF-8 data:
+
+~~~ sql
+SELECT '\xa7'::BYTEA::TEXT;
+pq: invalid UTF-8: "\xa7"
```
Comment (jseldess) [inline v1.0/porting-postgres.md]: Same as above.

File: v1.1/porting-postgres.md [hunk 3]
```diff
@@ -96,3 +96,36 @@ In CockroachDB, no such modulo is performed.
...
+----------------------------
+ \x48656c6c6f20776f726c6421
+~~~
+
+In CockroachDB, this conversion is performed by attempting to convert the
+byte data directly into a string:
+
+~~~ sql
+SELECT '\x48656c6c6f20776f726c6421'::BYTEA::TEXT;
++---------------------------------------------+
```
Comment (jseldess) [inline v1.1/porting-postgres.md]: Same as above.

File: v1.1/porting-postgres.md [hunk 4]
```diff
@@ -96,3 +96,36 @@ In CockroachDB, no such modulo is performed.
...
+(1 row)
+~~~
+
+A consequence of this is that such conversions will fail if the byte sequence
+contains invalid UTF-8 data:
+
+~~~ sql
+SELECT '\xa7'::BYTEA::TEXT;
+pq: invalid UTF-8: "\xa7"
```
Comment (jseldess) [inline v1.1/porting-postgres.md]: Same as above.

**Comment by justinj** [comment]
Thanks for the tip @jseldess, updated!

File: v1.0/porting-postgres.md [hunk 5]
```diff
@@ -96,3 +96,42 @@ In CockroachDB, no such modulo is performed.
...
+~~~
+
+In CockroachDB, this conversion is performed by attempting to convert the
+byte data directly into a string:
+
+~~~ sql
+SELECT '\x48656c6c6f20776f726c6421'::BYTEA::TEXT;
+~~~
+
+~~~ sql
```
Comment (jseldess) [inline v1.0/porting-postgres.md]: One more nit that I should've been clearer about: For responses, we leave the language unspecified, so can you remove `sql` here and for the other response blocks?

**Comment by justinj** [comment]
Ah, of course! Fixed.

**Comment by justinj** [comment]
If you wouldn't mind, hold off on merging this for the time being @jseldess, I want to have some discussion on #17969 regarding this particular issue.

**Comment by jseldess** [comment]
Sure thing, @justinj.

**Comment by jseldess** [comment]
@justinj, any updates on whether we want to merge this PR?

**Comment by justinj** [comment]
I'm working on changing the actual behaviour in https://github.com/cockroachdb/cockroach/pull/18843, running into some problems but should be able to work them out. I'll close this.

**Comment by jseldess** [comment]
Thanks, @justinj.

**Pull request closed without merging**